lib/Simulator/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+set(LLVM_OPTIONAL_SOURCES
+    State.cpp
+    Engine.cpp
+    signals-runtime-wrappers.cpp
+)
+
+add_mlir_library(CIRCTLLHDSimState
+    State.cpp
+
+    LINK_LIBS PUBLIC
+    LLVMSupport
+    )
+
+add_mlir_library(circt-llhd-signals-runtime-wrappers SHARED
+    signals-runtime-wrappers.cpp
+
+    LINK_LIBS PUBLIC
+    CIRCTLLHDSimState
+)
+
+add_mlir_library(CIRCTLLHDSimEngine
+    Engine.cpp
+
+    LINK_LIBS PUBLIC
+    MLIRLLHD
+    MLIRLLHDToLLVM
+    CIRCTLLHDSimState
```

Comment:
Danger: this gets included through botha a static and dynamic loading path.  This is generally a bad idea.

Reply:
I'm not sure what's the best way to solve this. Would it be fine to either remove it from here and leave it only in the `circt-llhd-signals-runtime-wrapper` library, or changing `circt-llhd-signals-runtime-wrappers` to have `LINK_LIBS PRIVATE`?

Reply:
What are you trying to accomplish with the shared library?

Reply:
The objective is to make the runtime functions from that library available when they are called from the LLVM dialect functions, that are invoked through the `mlir::ExecutionEngine` (in the `CIRCTLLHDSimEngine` library) during simulation

Reply:
OK, I'm not quite sure how to solve this at the moment.  I think this is something that we can deal with after the merge, if it's not causing problems up front.